pkg/tfgen/docs.go

```diff
@@ -2358,3 +2361,53 @@ var (
 func guessIsHCL(code string) bool {
 	return guessIsHCLPattern.MatchString(code)
 }
+
+func plainDocsParser(docFile *DocFile, pkgName string, g *Generator) ([]byte, error) {
+	// Replace upstream front matter with Pulumi registry's
+	contentStr, err := replaceUpstreamFrontMatter(string(docFile.Content), pkgName)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: translate code blocks
+
+	//TODO: apply default edit rules
+
+	//TODO: reformat text
+
+	//TODO: Light translation / possible eliding for certain headers such as "Arguments Reference"
+	// or "Configuration block"
+	return []byte(contentStr), nil
+}
+
+func replaceUpstreamFrontMatter(content, pkgName string) (string, error) {
+	// Capitalize the package name
+	capitalize := cases.Title(language.English)
+	pkgName = capitalize.String(pkgName)
```

Comment:
Shouldn't this be the `DisplayName` field on the schema?

Reply:
It should. But we do not in fact enforce this field on our providers. I'll add some logic to use DisplayName preferentially.